src/components/pagination/pagination.js

```diff
@@ -0,0 +1,150 @@
+			showItemCountSelect : { type: Boolean, attribute: 'show-item-count-select', reflect: true },
+			itemCountOptions : { type: Array, attribute: 'item-count-options' },
+			selectedCountOption : { type: Number, attribute: 'selected-count-option', reflect: true },
+		};
+	}
+
+	static get styles() {
+		return [selectStyles, css`
+			:host {
+				align-items: center;
+				display: flex;
+				justify-content: center;
+				white-space: nowrap;
+			}
+
+			.d2l-page-selector-container {
+				margin: 0.75rem;
+			}
+
+			.d2l-page-number-container {
+				direction: ltr;
+				display: inline-block;
+			}
+
+			.d2l-page-number {
+				margin-left: 0.25rem;
+				margin-right: 0.25rem;
+				width: 3.8rem;
+			}
+
+			.d2l-page-max-text {
+				margin-right: 0.25rem;
+			}
+
+			@media (max-width: 544px) {
+				:host {
+					flex-direction: column-reverse;
+				}
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.pageNumber = 1;
+		this.maxPageNumber = 1;
+		this.itemCountOptions = [10, 20, 30, 40];
+	}
+
+	render() {
+		return html`
+			<div class="d2l-page-selector-container">
+				<d2l-button-icon id="d2l-labs-pagination-previous-button" icon="tier1:chevron-left" @click="${this._onPreviousClicked}" text="${this.localize('components:pagination:previousPage')}" ?disabled=${this._isFirstPage()}></d2l-button-icon>
+
+				<div class="d2l-page-number-container">
+					<d2l-input-number
+						class="d2l-page-number"
+						autocomplete="off"
+						label="${this.localize('components:pagination:currentPage', { pageNumber: this.pageNumber, maxPageNumber: this.maxPageNumber })}"
+						label-hidden
+						max-fraction-digits="0"
+						value="${this.pageNumber}"
+						@change=${this._onPageNumberChanged}>
+					</d2l-input-number>
+					<!-- Note: this uses a division slash rather than a regular slash -->
+					<!-- a11y note: setting aria-hidden to true because info here is covered by input element -->
+					<span class="d2l-page-max-text" aria-hidden="true">∕ ${this.maxPageNumber}</span>
+				</div>
+				<d2l-button-icon id="d2l-labs-pagination-next-button" icon="tier1:chevron-right" @click="${this._onNextClicked}" text="${this.localize('components:pagination:nextPage')}" ?disabled=${this._isLastPage()}></d2l-button-icon>
+			</div>
+
+			${this.showItemCountSelect ? html`
+				<select
+					class="d2l-input-select"
+					aria-label="${this.localize('components:pagination:resultsPerPage')}"
+					title="${this.localize('components:pagination:resultsPerPage')}"
+					@change="${this._onPageCounterChanged}">
+					${this.itemCountOptions.map(item => html`
+						<option ?selected="${this.selectedCountOption === item}" value="${item}">${this.localize('components:pagination:amountPerPage', 'count', item)}</option>
+					`)}
+				</select>
+			` : null }
+		`;
+	}
+
+	_dispatchPageChangeEvent(newPageNumber) {
+		this.dispatchEvent(
+			new CustomEvent('d2l-labs-pagination-page-change', {
+				detail: { page: newPageNumber },
+				bubbles: true,
```

Comment:
Again, not sure how disruptive this would be, but normally we would set `bubbles: false` and `composed: false`. We generally only set them to true if we know it will be necessary, especially `composed`. If we need to maintain to avoid breakage, that's fine.

Reply:
We talked about it quickly over Slack, but I'll mention it here as well in case others see this. I'm deciding to not make this change yet, since I don't know how it'll impact the current uses of the component once it's changed. So the plan is to update everywhere first, then return to here and update to see if anywhere is impacted or not.